test/extract-js.js

```diff
@@ -0,0 +1,71 @@
+var assert = require('assert');
+var Checker = require('../lib/checker');
+
+describe('modules/extract', function() {
+    describe('common', function() {
+        it('should not throw an exceptions on error explain (console reporter)', function() {
+            var checker = new Checker();
+            checker.registerDefaultRules();
+            checker.configure({
+                disallowKeywords: ['with'],
+                extract: 'always'
+            });
+
+            return checker.checkPath('./test/data/extract').then(function(errorsCollection) {
+                var errorCount = 0;
+
+                assert.doesNotThrow(function() {
+                    errorsCollection.forEach(function(errors) {
+                        if (!errors.isEmpty()) {
+                            errors.getErrorList().forEach(function(error) {
+                                errors.explainError(error, true);
+                                errorCount++;
+                            });
+                        }
+                    });
+                });
+
+                assert(errorCount === 4);
+            });
+        });
+    });
+
+    describe('whitespaces', function() {
+        it('should be no errors on disallowTrailingWhitespace', function() {
+            var checker = new Checker();
+            checker.registerDefaultRules();
+            checker.configure({
+                disallowTrailingWhitespace: true
+            });
+            return checker.extractFile('./test/data/extract/index.html').then(function(errors) {
+                assert(errors === null);
+            });
+        });
+
+        it('should be errors on disallowMixedSpacesAndTabs', function() {
+            var checker = new Checker();
+            checker.registerDefaultRules();
+            checker.configure({
+                disallowMixedSpacesAndTabs: true
+            });
+            return checker.extractFile('./test/data/extract/index.html').then(function(errors) {
+                // disallowMixedSpacesAndTabs doesn't warn when some line contains only and some
+                // only spaces (i.e. '\s\svar foo;\t\tvar bar;'). If this behavior will change
```

Comment:
Well, I expect that if some lines has tabs offset but others spaces offset – warning should be shown. For now nothing happen. So, here is we will see 2 warnings:

```
\s\t
\t\s
```

But in this case no warnings:

```
\s\s
\t\t
```

But probably it's a feature, not a bug ;)